docs/README.md

```diff
@@ -34,6 +34,55 @@ You'll need a `kubectl`-style config file to connect to your cluster.
 `$HOME/.kube/config`. You can specify an alternate kubeconfig connection file
 with `--kubeconfig`, or the env var `$KUBECONFIG`, for any command.
 
+## Kn Config
+
+To better customize Kn there is a small set of configuration parameters you are able to setup. In particular you can specify where your Kn plugins are located and how they are found. The Kn configuration file is meant to capture these limited configuration options. Let's explore this file, it's location, and the current options you are able to change with it.
+
+### Location
+
+By default the Kn config file is located in the user's home directory under `.kn` hidden subdirectory and is named `config.yaml`. So in `~/.kn/config.yaml`. It is not created for you by default. You can easily create one with:
+
+```bash
+mkdir ~/.kn
+touch ~/.kn/config.yaml
+```
+
+Once that file exists (even when empty) Kn will indicate that is being used when you invoke a command. And default values for the configurable parameters will be added (if they were not present). For instance:
+
+```bash
+./kn plugin list
+Using kn config file: /home/maximilien/.kn/config.yaml
+No plugins found.
+```
+
+To see the values in the Kn config file you can do:
+
+```bash
+cat ~/.kn/config.yaml 
+lookuppluginsinpath: false
+pluginsdir: ~/.kn/plugins
+```
+
+You can now modify these values by editing the file. To specify a different config file then use the `--config` persistent flag when invoking Kn.
+
+### Options
+
+Currently there are two options you can specific in the Kn config file and they are related to how Kn locates plugins.
+
+1. `pluginsDir` or `pluginsdir` which is the same as the persistent flag `--plugins-dir` and  specifies the Kn plugins directory. It defaults to: `~/.kn/plugins`. By using the persistent flag (when you issue a command) or by specifying the value in the Kn config, a user can selects which directory to find Kn plugins. It can be any directory that is visible to the user.
+
+2. `lookupPluginsInPath` or `lookuppluginsinpath` which is the same as the persistent flag `--lookup-plugins-in-path` and specficies if Kn should look for plugins anywhere in the currently specified `PATH` shell variable. The value for this configuration parameter is `true` or `false`.
+
+For example, the following Kn config will look for Kn plugins in the user's `PATH`.
+
+```bash
+cat ~/.kn/config.yaml
+lookupPluginsInPath: true
+pluginsdir: ~/.kn/plugins
+```
+
+Note that in the configuration above, you need to add `~/.kn/plugins` directory in your `PATH` variable if you want plugins located in that directory to be seen and be usable with Kn.
```

Comment:
Is this really true ? I read the configuration above as lookup in `~/.kn/plugins` *AND* in the PATH. So no need to put `~/.kn/plugins` also into the path. At least that is what I would suppose as a user (because otherwise there is no need to add `pluginsDir` at all).

Reply:
I don't think so. Will verify.

Reply:
Actually correct. It does find and execute plugins in `~/.kn/plugins` and I think behavior I am seeing is based on issue #417. My Linux system is running Ubuntu but I doubt that's the issue. I need to investigate what's going on. But will do that when I address #417. 

Fixing now since the behavior you mention is ideal.

Reply:
Also the overall behavior is broken (independently of #417) since on a Mac OS X system I am seeing that if I have a `~/.kn/config.yaml` it does get populated but if I change `lookupPluginsInPath: true` and combination of `pluginsDir` I am not getting the right thing. 

I need to get add some integration / e2e tests for plugins and get to the bottom of this. 

OK updating this PR but lots to get done to have this doc be what the code is doing. Feel free to lgtm if all otherwise look good since it reflects what we want.